driver-sync/src/test/functional/com/mongodb/client/unified/ContextElement.java

```diff
@@ -467,4 +508,40 @@ private static BsonDocument serverMonitorEventToDocument(final Object event) {
         return new BsonDocument(EventMatcher.getEventType(event.getClass()),
                 new BsonDocument("awaited", BsonBoolean.valueOf(EventMatcher.getAwaitedFromServerMonitorEvent(event))));
     }
+
+    static BsonDocument topologyEventToDocument(final Object event) {
+        if (event != null && !(event instanceof ClusterOpeningEvent || event instanceof ClusterDescriptionChangedEvent || event instanceof ClusterClosedEvent)) {
+            throw new UnsupportedOperationException("Unsupported topology event: " + event.getClass().getName());
+        }
+        BsonDocument eventDocument = new BsonDocument();
+        if (event instanceof ClusterDescriptionChangedEvent) {
+            ClusterDescriptionChangedEvent changedEvent = (ClusterDescriptionChangedEvent) event;
+            eventDocument.put("previousDescription",
+                    new BsonDocument("type", new BsonString(clusterDescriptionToString(changedEvent.getPreviousDescription()))));
+            eventDocument.put("newDescription",
+                    new BsonDocument("type", new BsonString(clusterDescriptionToString(changedEvent.getNewDescription()))));
+        }
+        return new BsonDocument(EventMatcher.getEventType(event.getClass()), eventDocument);
+    }
+
+    static String clusterDescriptionToString(final ClusterDescription clusterDescription) {
+        switch (clusterDescription.getType()) {
+            case STANDALONE:
+                return "Single";
+            case REPLICA_SET:
+                return clusterDescription.getServerDescriptions().stream().
+                        filter(ServerDescription::isPrimary)
+                        .findFirst()
+                        .map(serverDescription -> "ReplicaSetWithPrimary")
+                        .orElse("ReplicaSetNoPrimary");
```

Comment:
the above stream usage works but is not idiomatic (and less verbose) as below:
```
case REPLICA_SET:
                return clusterDescription.getServerDescriptions().stream().
                        anyMatch(ServerDescription::isPrimary) ? "ReplicaSetWithPrimary" : "ReplicaSetNoPrimary";
```

Reply:
Thats neater :) - done